src/utils/index.ts

```diff
@@ -79,7 +79,7 @@ export const pgSnapshotReadWrapper =
   (handlerFactory: (pool: PoolOrClient) => Handler) =>
   (pool: Pool): Handler =>
   async (req: Request, res: Response, next: NextFunction) => {
-    pgSnapshotRead(pool, async (client) =>
+    await pgSnapshotRead(pool, async (client) =>
```

Comment:
This is the only fix needed. Awaiting the promise to resolve to avoid **unhandled promise rejection error**